src/wayland/shell/legacy/mod.rs

```diff
@@ -64,6 +64,7 @@ use super::PingError;
 /// The role of a wl_shell_surface.
 pub const WL_SHELL_SURFACE_ROLE: &str = "wl_shell_surface";
 
+pub mod decoration;
```

Comment:
I think this should be `kde_decoration` to make it clear that it is kde's decoration protocol.

Reply:
Oh yeah, that's definitely one thing I wasn't sure about: Where to put it.

It's legacy, but it's not part of the legacy wl_shell. Could create a `decoration` module in the root but then I'd expect the XDG protocol to be found there too.

Reply:
I would put this under `wayland/shell/kde_decoration`.

As long as there are client, that are actively using it, it is (sadly) not legacy. And it belongs to shells, but is not directly related to either `wl_shell` or `xdg_shell`.

Reply:
How would you feel about `shell/kde/decoration` to match the `shell/xdg/decoration`, which could also be used for other KDE-created protocols? I'm not sure if there are any other KDE protocols or if there will be in the future, but it's definitely an option.

Reply:
Well according to [wayland.app](https://wayland.app/protocols/) there are many. I am not sure, if we want to support a lot of them, but a `kde` module will certainly make this easier later.